pymeasure/instruments/mksinst/mks937b.py

```diff
@@ -127,13 +177,11 @@ def __init__(self, adapter, name="MKS 937B vacuum gauge controller", address=253
 
     unit = Instrument.control(
         "U?", "U!%s",
-        """Control pressure unit used for all pressure readings from the instrument""",
+        """Control pressure unit used for all pressure readings from the instrument.
+
+        Allowed units are Unit.Torr, Unit.mbar, Unit.Pa, Unit.uHg""",
         validator=strict_discrete_set,
         map_values=True,
-        values={"Torr": "TORR",
-                "mBar": "mBAR",
-                "Pascal": "PASCAL",
-                "Micron": "MICRON",
-                },
+        values={u: u.value for u in Unit},
```

Comment:
I think, that we should allow the old values as well in order to be backward compatible.

Reply:
I added back the old values in addition to the new ones. This solves setting the unit with the old string option. Still upon reading the new `Unit` enum is returned since I kept those at the top of the dictionary. It could be swapped to keep the old as default, but honestly I consider the enum the better solution and it just shifts the breaking change to a later point. Any ideas?

Reply:
I see. This instrument has been released, already, didn't it?

If you make the enum an `enum.StrEnum` (or `class StrEnum(str, Enum)` for python < 3.11), it will behave like a string, such that it will be backward compatible.

Reply:
you can do
~~~python
try:
    from enum import StrEnum
except ImportError:
    from enum import Enum
    class StrEnum(str, Enum):
        pass

Reply:
Yes the MKS937B was already part of the last published pymeasure version.

I now use StrEnum for both devices, but it is not backward compatible. I believe full backward compatibility can't be maintained while also moving to the enum. Personally I consider it worth breaking the backward compatibility in favor of moving to the enum which is more easy to use.

(Why is it not backward compatible: previously the user could set `dev.unit = 'Torr'`, even with `StrEnum` it is now `dev.unit = "TORR"`. Its even worse with `mbar` where the device requires a mix small/captital spelling of `mBAR`)

I am not sure what your policy is with respect to backward compatibility. Let me know how to proceed.

Reply:
The StrEnum makes the return value backward compatible.

For the accepted values, however, we should do it this way to include both possibilities:
~~~python
unit_dict = {u: u.value for u in Units}
unit_dict.update({
    "Torr": "TORR",
    # ...
})
~~~

Reply:
I believe I am missing something. The return value is still `Unit.Torr` which also as StrEnum equals only "TORR" but not "Torr". So the backward compatibility issue which is difficult to solve is for reading. What you suggest and what I had before solves only the writing. To have it backward compatible for both reading and writing one would need to use the old values first. Then however the enum can only be used for writing. See [1dad20a](https://github.com/pymeasure/pymeasure/pull/1034/commits/1dad20ab85ca80e53dcc1658e140418d00190544)